kv/keyflags.go

```diff
@@ -32,6 +34,26 @@ func (f KeyFlags) HasNeedLocked() bool {
 	return f&flagNeedLocked != 0
 }
 
+// HasAssertExists returns whether the key is asserted to already exist before the current transaction.
+func (f KeyFlags) HasAssertExists() bool {
+	return f&flagAssertExists != 0 && f&flagAssertNotExists == 0
+}
+
+// HasAssertNotExists returns whether the key is asserted not to exist before the current transaction.
+func (f KeyFlags) HasAssertNotExists() bool {
+	return f&flagAssertNotExists != 0 && f&flagAssertExists == 0
+}
+
+// HasAssertUnknown returns whether the key is unable to do any assertion.
+func (f KeyFlags) HasAssertUnknown() bool {
+	return f&flagAssertExists != 0 && f&flagAssertNotExists != 0
```

Comment:
When will the two flags both exist at the same time? This is a bit confusing. How about use a single flag `flagAssertUnknown`?

Reply:
There are four states: None, AssertExist, AssertNotExist, Unknown, and can be represented by at least two bits. lysu's design is to use both zero to represent None, and both one to represent unknown. I think I can add more comments to explain it.